lib/schemaUtils.js

```diff
@@ -1887,16 +1888,19 @@ module.exports = {
       }
 
       // header found in spec. check header's schema
-      process.nextTick(() => {
-        this.checkValueAgainstSchema(mismatchProperty,
-          transactionPathPrefix + '[?(@.key==\'' + pHeader.key + '\')]',
-          pHeader.value,
-          schemaHeader.pathPrefix + '[?(@.name==\'' + schemaHeader.name + '\')]',
-          deref.resolveRefs(schemaHeader.schema, 'request', components),
-          components, options,
-          cb
-        );
-      });
+      // Need a closure here as the schemaHeader value was getting replaced to undefined
+      ((schemaHeader) => {
```

Comment:
I was getting `Cannot read property pathPrefix of undefined`. Since there is a bailout just above which checks for this, the behavior seemed unexpected.